File: API/AuthService/Models/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace AuthService.Models
{
    // Ця DTO підійде для реєстрації і клієнта, і офіціанта
    public class RegisterRequest
    {
        [Required]
        public string Username { get; set; }

        [Required]
        [MinLength(6)]
        public string Password { get; set; }

        public string? Email { get; set; } // Використовується тільки для клієнта
        public string? Phone { get; set; } // Використовується тільки для клієнта
        public string? FullName { get; set; } // Використовується тільки для офіціанта
    }
}

File: ClientSide/UkrainianStyleRestaurant_2.0_another/app/src/main/java/com/example/ukrainianstylerestaurant/model/RegisterRequest.java
package com.example.ukrainianstylerestaurant.model;

public class RegisterRequest {
    public String username;
    public String password;
    public String email;
    public String phone;

    public RegisterRequest(String username, String password, String email, String phone) {
        this.username = username;
        this.password = password;
        this.email = email;
        this.phone = phone;
    }
}